### PR TITLE
Upd conda recipe

### PR DESCRIPTION
**Pull request by k-dominik**
Summary:

* Added a "first" conda recipe. setup.py is currently still source of truth for dependencies/constraints. Used run_constrained section to specify versions for optional packages.
* Added some installation instructions
* Added basic ci

Todo:

* [ ] come up with sensible minimal (max) requirements for onnxruntime, tensorflow (are there additional dependencies that go with those?)
* [x] add ci for various configs

supersedes #79

### CODE REVIEW

**Comment by k-dominik** [comment]
hmm I guess this package relies on an unreleased version of `bioimageio.spec` (which seems not to be compatible with the latest current release). Of course we could start relying on the main branch of `bioimageio.spec` or adding it as a submodule. Distributing `bioimageio.core` independently doesn't make a lot of sense then (imo), with outdated upstream dependency (`bioimageio.spec`).

CC: @FynnBe @constantinpape

**Comment by constantinpape** [comment]
> Edit: Alternatively we could just add more "configurations" here. Once using `bioimageio.spec` `main`, and once with the latest package.

Yes, I think that would be my preferred solution.

**Comment by k-dominik** [comment]
alright, this can be reviewed - note:

test failures against conda `bioimage.spec` package are expected. Added tests against `main`.
Current conda recipe will of course not work, as `bioimageio.spec` is outdated on conda-forge.

File: README.md [hunk 1]
```diff
@@ -1,6 +1,37 @@
...
-# Model Specification
-The model specification and its validation tools have moved to https://github.com/bioimage-io/spec-bioimage-io.
+## Installation
+
+The `bioimageio.core` package supports various back-ends for running BioimageIO networks:
+
+* Pytorch/Torchscript:
+  ```bash
+  # cpu installation (if you don't have an nvidia graphics card)
+  conda install -c pytorch -c conda-forge -c ilastik-forge bioimageio.core pytorch torchvision cpuonly
```
Comment (constantinpape) [inline README.md:12]: Why do we need `ilastik-forge` in the channels here? Same for the conda install commands below.

File: dev/environment-base.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+name: bio-core-dev
+channels:
+  - pytorch
+  - conda-forge
+  - defaults
+dependencies:
```
Comment (constantinpape) [inline dev/environment-base.yaml:6]: Should we add pytorch in the dependencies? Or remove the pytorch channel?

**Review by constantinpape — COMMENTED** [review]
Some minor comments, looks good otherwise.
I would vote for merging this despite the failing tests, so that we can check that things work with spec main.

File: README.md (hunk 1 above)
Comment (k-dominik) [reply README.md:12]: Not really once we have the package on conda-forge. Until then we could have some versions on ilastik-forge. Not a very strong opinion there. But this makes it easier to remove certain versions in the beginning. (I of course know of such mechanisms in the conda-forge community, but would not want to put too much stress on the community. I'd prefer to come to a somewhat stable api first).

File: dev/environment-base.yaml (hunk 2 above)
Comment (k-dominik) [reply dev/environment-base.yaml:6]: right, in principle we should test

* no NN backend
* Pytorch
* onnxruntime
* tensorflow2
?

**Comment by k-dominik** [comment]
> Some minor comments, looks good otherwise.
> I would vote for merging this despite the failing tests, so that we can check that things work with spec main.

okay I think we can potentially merge this. This will test with a single (for each python version) environment (cpuonly) with `onnxruntime`, `pytorch`, `tensorflow`.

I guess we can add pins/minimal versions to those later when needed.

**Comment by constantinpape** [comment]
Thanks Dominik!

**Pull request merged**